src/main/java/frc/robot/utils/SimmableCANSparkMax.java

```diff
@@ -50,6 +59,78 @@ public CANError setReference(double value, ControlType ctrl, int pidSlot, double
 
   }
 
+  private class SimmableCANDigitalInput extends CANDigitalInput {
+    private LimitSwitch type;
+    private SimBoolean getter;
+
+    public SimmableCANDigitalInput(CANSparkMax device, LimitSwitch limitSwitch, LimitSwitchPolarity polarity) {
+      super(device, limitSwitch, polarity);
+
+      type = limitSwitch;
+    }
+
+    @Override
+    public CANError enableLimitSwitch(boolean enable) {
+      if (m_simDevice != null && enable) {
+        if (type == LimitSwitch.kForward) {
+          if (m_fwdLimitSwitch == null) {
+            m_fwdLimitSwitch = m_simDevice.createBoolean("FWD Limit Switch", false, false);
+          }
+          getter = m_fwdLimitSwitch;
+        } else {
+          if (m_rvsLimitSwitch == null) {
+            m_rvsLimitSwitch = m_simDevice.createBoolean("RVS Limit Switch", false, false);
+          }
+          getter = m_rvsLimitSwitch;
+        }
+      }
+
+      return super.enableLimitSwitch(enable);
+    }
+
+    @Override
+    public boolean get() {
+      if (getter != null) {
+        return getter.get();
+      } else {
+        return super.get();
+      }
+    }
+
+  }
+
+  private class SimmableCANEncoder extends CANEncoder {
+
+    public SimmableCANEncoder(CANSparkMax device, EncoderType sensorType, int counts_per_rev) {
+      super(device, sensorType, counts_per_rev);
+    }
+
+    @Override
+    public double getPosition() {
+      if (m_simDevice != null) {
+        if (m_encoderCount == null) {
+          m_encoderCount = m_simDevice.createDouble("Encoder Pos", false, 0);
+        }
+        return m_encoderCount.get();
+      } else {
+        return super.getPosition();
+      }
+    }
+
+    @Override
+    public double getVelocity() {
+      if (m_simDevice != null) {
+        if (m_encoderCount == null) {
+          m_encoderCount = m_simDevice.createDouble("Encoder Vel", false, 0);
+        }
+        return m_encoderCount.get();
+      } else {
+        return super.getPosition();
```

Comment:
Should return super.getVelocity().

Reply:
And there it is.  Thanks so much for tracking that one down.

Reply:
Ah, whoops. That one's on me